src/main/java/dev/ftb/extendedexchange/EMCSyncHandler.java

```diff
@@ -0,0 +1,46 @@
+package dev.ftb.extendedexchange;
+
+import moze_intel.projecte.api.capabilities.PECapabilities;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.server.ServerLifecycleHooks;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * Track players who need their EMC value sync'd, and do it in a controlled fashion once a second instead of every time
+ * an EMC link block does anything. Avoids excessive sync'ing and performance issues with many such blocks in the world.
+ */
+public enum EMCSyncHandler {
+    INSTANCE;
+
+    private final Set<UUID> toSync = new HashSet<>();  // UUID of players who need EMC sync'd
+    private int counter = 0;
+
+    public void needsSync(ServerPlayer player) {
+        toSync.add(player.getUUID());
+    }
+
+    public void needsSync(UUID playerId) {
+        toSync.add(playerId);
+    }
+
+    public void onServerTick(TickEvent.ServerTickEvent event) {
+        if (event.phase == TickEvent.Phase.END) {
+            counter++;
+            if (counter == 20) {
+                MinecraftServer server = ServerLifecycleHooks.getCurrentServer();
```

Comment:
Is this the best way to get the server? I know this method can cause issues in some rare cases. I can't imagine this would cause issues, but does the event not hold the server?

Reply:
Not in 1.18.  `getServer()` was only added in 1.19 - surprised me too. Should be safe to use in a server tick event though - the event can't be fired unless there's a server to fire it :)